demos/tutorials/multiple_likelihoods.py

```diff
@@ -0,0 +1,268 @@
+"""
+# %% 
+# First we import the python libraries needed.
+import sys
+sys.path.append("../../")
+import cuqi
+import numpy as np
+import matplotlib.pyplot as plt
+from math import ceil
+
+# %%
+# Choose one of the two cases we study in this demo 
+# -------------------------------------------------
+#
+# We can choose between two cases:
+# Choose `set_up = set_ups[0]` for the case where we have two 1D Poisson models
+# that differ in the observation operator only. And choose `set_up = set_ups[1]`
+# for the case where we have two 1D Poisson models that differ in the source 
+# term only. Here we demonstrate the first case, `set_up = set_ups[0]`.
+
+set_ups = ["multi_observation", "multi_source"]
+set_up = set_ups[0]
+
+assert set_up == "multi_observation" or set_up == "multi_source",\
+    "set_up must be either 'multi_observation' or 'multi_source'"
+
+# %%
+# Set up the parameters used in both models
+# -----------------------------------------
+
+dim = 50  # Number of the model grid points
+endpoint = 1  # The model domain is the interval [0, endpoint]
+field_type = "Step"  # The conductivity (or diffusivity) field type.
+                     # We choose step function parameterization here.
+SNR = 400  # Signal-to-noise ratio
+n_steps = 2  # Number of steps in the conductivity (or diffusivity) step field
+magnitude = 100 # Magnitude of the source term in the Poisson problem
+
+# Exact solution
+x_exact = np.empty(dim)
+x_exact[:ceil(dim/2)] = 2
+x_exact[ceil(dim/2):] = 3
+
+
+# %%
+# Set up the first model
+# ----------------------
+#
+# We set up the first forward model to have observations at the first half of 
+# the domain (or observation everywhere if `set_up = set_ups[1]`). We then plot
+# the true conductivity field (the exact solution), the exact data and the noisy
+# data.
+
+observation_grid_map1 = None
+if set_up == "multi_observation":
+	# Observe on the first half of the domain
+	observation_grid_map1 = lambda x: x[np.where(x<.5)] 
+
+# The source term signal
+source1 = lambda xs: magnitude*np.sin(xs*2*np.pi/endpoint)+magnitude
+
+# Obtain the forward model from the test problem
+model1, data1, problemInfo1 = cuqi.testproblem.Poisson_1D.get_components(dim=dim,
+                                                                         endpoint=endpoint,
+                                                                         field_type=field_type,
+                                                                         field_params={
+                                                                             'n_steps': n_steps},
+                                                                         observation_grid_map=observation_grid_map1,
+                                                                         exactSolution=x_exact,
+                                                                         source=source1,
+                                                                         SNR=SNR)
+
```

Comment:
This does not render nicely on the online docs as the line is too long. I would suggest using the black-style formatting.

We should set up a GitHub action to automtically apply black formatting, but until then you can use http://black.vercel.app
```suggestion
model1, data1, problemInfo1 = cuqi.testproblem.Poisson_1D.get_components(
    dim=dim,
    endpoint=endpoint,
    field_type=field_type,
    field_params={"n_steps": n_steps},
    observation_grid_map=observation_grid_map1,
    exactSolution=x_exact,
    source=source1,
    SNR=SNR,
)

```